crates/ruff_linter/src/rules/ruff/rules/redirected_noqa.rs

```diff
@@ -67,3 +67,32 @@ pub(crate) fn redirected_noqa(diagnostics: &mut Vec<Diagnostic>, noqa_directives
         }
     }
 }
+
+/// RUF101 for file noqa directives
+pub(crate) fn redirected_file_noqa(
+    diagnostics: &mut Vec<Diagnostic>,
+    noqa_directives: &FileNoqaDirectives,
+) {
+    for line in noqa_directives.lines() {
+        let ParsedFileExemption::Codes(codes) = &line.parsed_file_exemption else {
+            continue;
+        };
+
+        for code in codes.iter() {
+            if let Some(redirected) = get_redirect_target(code.as_str()) {
+                let mut diagnostic = Diagnostic::new(
+                    RedirectedNOQA {
+                        original: code.to_string(),
+                        target: redirected.to_string(),
+                    },
+                    code.range(),
+                );
+                diagnostic.set_fix(Fix::safe_edit(Edit::range_replacement(
+                    redirected.to_string(),
+                    code.range(),
+                )));
+                diagnostics.push(diagnostic);
+            }
+        }
+    }
```

Comment:
Hmm, it's annoying that `FileNoqaDirectives` and `NoqaDirectives` have such a differetn structure. But maybe there's an opportunity to reuse some of the code in this loop?

Reply:
Oh yes, great idea! I factored this out into a `build_diagnostics` function.